fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -937,6 +950,19 @@ protected CompletableFuture<Void> iterateAllRanges(List<Object> additionalLogMes
                 getRunner().getExecutor());
     }
 
+    public CompletableFuture<Void> mergeIndexes() {
+        return AsyncUtil.whenAll(luceneIndexes.stream()
+                .map(index2 -> getRunner().runAsync(context -> openRecordStore(context).thenApply(store ->
+                    store.getIndexMaintainer(index2).mergeIndex().join()
+                ))).collect(Collectors.toList()));
+    }
```

Comment:
This seems like it exposes too many details for other adopters to use. A better way would be to create a new `IndexOperation` to be used here.

Perhaps there's something more opaque we could provide that would serve these needs, but allow the user to avoid having to know what types of maintenance they are deferring (if that's even a good idea).
Something like a an option the `FDBRecordStore`: `allowDeferredIndexMaintenance`, `needsDeferredIndexMaintenance` and `performDeferredIndexMaintenance`.

Reply:
Can that be a future enhancement? 
The functions `getIndexMergePolicy` and `shouldPreventLuceneMergeDuringIndexing` are marked as experimental, will that allow us future changes?